### PR TITLE
Add a python faucet

### PR DESCRIPTION
**Pull request by measure-fi**
https://www.notion.so/volumefi/EPIC-06-17-Paloma-Public-Testnest-1-d11ce23143c7475a9157acd224933d81

# Background

Temporary super hacky faucet

# Testing completed

- [x] Manually tested

### CODE REVIEW

File: pyfaucet/faucet.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from bottle import post, request, run
+import sqlite3
+import subprocess
+
+BANK = "paloma167rf0jmkkkqp9d4awa8rxw908muavqgghtw6tn"
+
+db = sqlite3.connect("faucet.db")
```
Comment (measure-fi) [inline pyfaucet/faucet.py:7]: BANK and db require preexisting assets on the node.

File: pyfaucet/faucet.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    if tx.stderr:
+        print(tx.stderr)
+        return "failed"
+    cur.execute("INSERT INTO addresses VALUES (?, date('now'))", (address,))
+    db.commit()
+
+    print(tx.stdout)
+    return ""
+
+run(host="localhost", port=8479)
```
Comment (measure-fi) [inline pyfaucet/faucet.py:28]: Can change this to 8080 and deploy!

File: pyfaucet/faucet.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    tx = subprocess.run(["palomad", "--fees", "200grain", "--node", "tcp://testnet.palomaswap.com:26657", "tx", "bank", "send", "-y", BANK, address, "100000grain"], capture_output=True)
+    if tx.stderr:
+        print(tx.stderr)
+        return "failed"
+    cur.execute("INSERT INTO addresses VALUES (?, date('now'))", (address,))
+    db.commit()
+
+    print(tx.stdout)
+    return ""
```
Comment (measure-fi) [inline pyfaucet/faucet.py:26]: Responses are at the moment almost nonsensical; tell me what the frontend wants and I can send it.

**Review by Vizualni — CHANGES_REQUESTED** [review]
I am very sorry, but I'll need to block this from getting merged into master. Please move this to a temporary repository instead.

**Comment by taariq** [comment]
Agreed this should be in its own repo

**Comment by measure-fi** [comment]
I don't necessarily think this should be merged anywhere, I'm hoping it is a very temporary stopgap. But I can drop it in another repo if we intend to do any maintenance on it (for instance, it is missing rate limiting of IPs and allowing people to go again).

**Comment by taariq** [comment]
It's good code. Please put it into its own repo. Others will surely use it.

**Comment by measure-fi** [comment]
https://github.com/palomachain/pyfaucet

**Pull request closed without merging**